### DISCUSSION TITLE
Unclear docs in "Viewing and Deleting Historical Data"

### DISCUSSION THREAD

**Discussion by refaelsh** [ryan4yin/nixos-and-flakes-book#188] [Q&A] [answered]
It is stated here https://nixos-and-flakes.thiscute.world/nixos-with-flakes/other-useful-tips#viewing-and-deleting-historical-data: `Due to the following issue, you need to run the gc command as per user to delete home-manager's historical data`.
I don't understand what it means. Does this mean I need to run `nix-collect-garbage --delete-old` as root and as the user?

Please explain this to me :-)

Thanks in advance.

**Comment by ryan4yin** [comment, marked as answer]
If you're not using home-manager, then you can safely ignore this comment and the command.

The issue is that, `sudo nix-collect-garbage --delete-old` will only delete all the system-level's historical data, and `sudo` means your're running the command under `root`.

But the home-manager's historical data will remains on your system, so you should run `nix-collect-garbage --delete-old` without `sudo`, to clear all the data for your current user.

You need to run `nix-collect-garbage --delete-old` under all users with home-manager installed , that's why I said `per user` in the docs.
But Generally speaking, you will only have home-manager under your personal account, so you don't need to run this command multiple times.

**Reply by refaelsh** [reply]
I understand. Thank you very much :-)
